castai/resource_rebalancing_job.go

```diff
@@ -0,0 +1,220 @@
+package castai
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"strings"
+	"time"
+
+	"github.com/castai/terraform-provider-castai/castai/sdk"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/samber/lo"
+)
+
+func resourceRebalancingJob() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceRebalancingJobCreate,
+		ReadContext:   resourceRebalancingJobRead,
+		DeleteContext: resourceRebalancingJobDelete,
+		UpdateContext: resourceRebalancingJobUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: rebalancingJobStateImporter,
+		},
+		Description: "Job assigns rebalancing shcedule to a cluster.",
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(1 * time.Minute),
+			Read:   schema.DefaultTimeout(1 * time.Minute),
+			Update: schema.DefaultTimeout(1 * time.Minute),
+			Delete: schema.DefaultTimeout(1 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			FieldClusterId: {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+				Description:      "CAST AI cluster id.",
+			},
+			"rebalancing_schedule_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+				Description:      "rebalancing schedule of this job",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "The job will only be executed if it's enabled",
+			},
+		},
+	}
+}
+
+func resourceRebalancingJobCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*ProviderConfig).api
+
+	job, err := stateToRebalancingJob(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	req := sdk.ScheduledRebalancingAPICreateRebalancingJobJSONRequestBody{
+		Id:                    job.Id,
+		ClusterId:             job.ClusterId,
+		RebalancingScheduleId: job.RebalancingScheduleId,
+		Enabled:               job.Enabled,
+	}
+
+	resp, err := client.ScheduledRebalancingAPICreateRebalancingJobWithResponse(ctx, *job.ClusterId, req)
+	if checkErr := sdk.CheckOKResponse(resp, err); checkErr != nil {
+		return diag.FromErr(checkErr)
+	}
+
+	d.SetId(*resp.JSON200.Id)
+
+	return resourceRebalancingJobRead(ctx, d, meta)
+}
+func resourceRebalancingJobRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
```

Comment:
Isn't there a linter configured for the provider? :thinking: Missing new line after the previous function.